src/index.js

```diff
@@ -1,7 +1,129 @@
-import UI from '../modules/UI.js';
 import './style.css';
 
-const ui = new UI();
+const taskInput = document.querySelector('.add-input');
+const taskBox = document.querySelector('.task-box');
 
-// trigger the renderTodos method to display the li
-ui.renderTodos();
+let editId;
+let isEditedTodo = false;
+
+// get data from the localstorage
+const todos = JSON.parse(localStorage.getItem('todo-list')) || [];
+
+function showTodos() {
+  taskBox.innerText = '';
+  todos.forEach((todo, index) => {
+    // apply the completed status iff the task/todo is completed
+    const isCompleted = todo.completed ? 'completed' : '';
+    const checked = isCompleted === 'completed' ? 'checked' : '';
+    taskBox.innerHTML += `
+            <li class="task">
+          <label for="${index}">
+            <input type="checkbox" ${checked} id="${index}" class="input"/>
+            <p class="${isCompleted}">${todo.description}</p>
+          </label>
+          <div class="actions">
+            <i class="fa-solid fa-ellipsis"></i>
+            <ul class="sub-actions">
+              <li data-info="${index}, ${todo.description}" class="edit"><i class="fa-solid fa-pen-to-square" id="edit"></i>Edit</li>
+              <li data-id="${index}" class="delete"><i class="fa-solid fa-trash-can" id="delete"></i>Delete</li>
+            </ul>
+          </div>
+        </li>  
+      `;
+  });
+}
+
+taskInput.addEventListener('keyup', (e) => {
+  const userTask = taskInput.value.trim();
+  if (e.key === 'Enter' && userTask) {
+    // fix the edit action
+    if (!isEditedTodo) {
+      const newTodo = {
+        description: userTask,
+        completed: false,
+        index: todos.length + 1,
+      };
+      // add the new todo to the todo list
+      todos.push(newTodo);
+    } else {
+      isEditedTodo = false;
+      todos[editId].description = taskInput.value;
+      window.location.reload();
+    }
+
+    taskInput.value = '';
+
+    // update the todo
+    localStorage.setItem('todo-list', JSON.stringify(todos));
+    showTodos();
+  }
+});
+
+showTodos();
+
+const editButtons = document.querySelectorAll('.edit');
+const deleteButtons = document.querySelectorAll('.delete');
+const toggleButtons = document.querySelectorAll('.fa-ellipsis');
+const statuses = document.querySelectorAll('.input');
+
+const showSubActions = (selectedTodo) => {
+  const taskMenu = selectedTodo.parentElement.lastElementChild;
+  taskMenu.classList.add('show');
+  document.addEventListener('click', (e) => {
+    if (e.target.tagName !== 'I' || e.target !== selectedTodo) {
+      taskMenu.classList.remove('show');
+    }
+  });
+};
+
+const editTask = (id, taskDescription) => {
+  editId = id;
+  isEditedTodo = true;
+  // put the todo description into the input field
+  taskInput.value = taskDescription;
+};
+
+function deleteTask(id) {
+  // remove one todo from the specified id
+  todos.splice(id, 1);
+  localStorage.setItem('todo-list', JSON.stringify(todos));
+  showTodos();
+}
```

Comment:
[See this Video](https://www.loom.com/share/bfd464fcf9114111b2f4ed1e67df754d)

- [ ] Great job so far 👏🏻 👏🏻  However as you just saw in the video above, I am unable to remove the book at least after reloading the page, and I must do this every time I need to remove the book . Please correct this.